fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java

```diff
@@ -96,5 +106,80 @@ public Void visitExecuteAsStatement(ExecuteAsStmt stmt, ConnectContext session)
             return null;
         }
 
+        @Override
+        public Void visitAlterUserStatement(AlterUserStmt stmt, ConnectContext session) {
+            analyseUserAndCheckExist(stmt.getUserIdent(), session);
+            /*
+             * IDENTIFIED BY
+             */
+            // convert password to hashed password
+            byte[] scramblePassword;
+            if (!Strings.isNullOrEmpty(stmt.getOriginalPassword())) {
+                try {
+                    // plain password should check for validation & reuse
+                    Auth.validatePassword(stmt.getOriginalPassword());
+                    GlobalStateMgr.getCurrentState().getAuth()
+                            .checkPasswordReuse(stmt.getUserIdent(), stmt.getOriginalPassword());
```

Comment:
Only plain password requires validation and reuse check. I suggest you put line 119 - 122 after line 123, just like the original code.